File: frontend/js/mockData.js
// Mock data for frontend development
const MockData = {
  // Sample analysis results for different scenarios
  goodAnalysis: {
    analysisId: "mock-good-001",
    timestamp: "2024-01-01T00:00:00Z",
    overallScore: 92,
    feedback: {
      summary:
        "Excellent work! You demonstrate strong understanding of calculus concepts and show clear problem-solving skills.",
      strengths: [
        "Clear step-by-step solutions",
        "Correct application of derivative rules",
        "Good use of mathematical notation",
        "Shows understanding of chain rule",
      ],
      weakAreas: [
        "Minor calculation errors in complex problems",
        "Could show more work for partial credit",
      ],
      recommendations: [
        "Double-check arithmetic in multi-step problems",
        "Practice more integration techniques",
        "Review limits and continuity concepts",
      ],
    },
    detailedAnalysis: {
      conceptUnderstanding: 95,
      problemSolving: 90,
      completeness: 88,
      accuracy: 92,
    },
    studyPlan: {
      priorityTopics: [
        "Integration by parts",
        "Trigonometric integrals",
        "Improper integrals",
      ],
      suggestedResources: [
        "Khan Academy Calculus II",
        "Paul's Online Math Notes",
        "Practice problems from textbook Chapter 7",
      ],
      estimatedTime: "3-4 hours",
      nextSteps: [
        "Complete integration practice set",
        "Review trigonometric identities",
        "Prepare for next exam",
      ],
    },
    confidence: 0.92,
  },

  averageAnalysis: {
    analysisId: "mock-average-001",
    timestamp: "2024-01-01T00:00:00Z",
    overallScore: 75,
    feedback: {
      summary:
        "Good progress overall. You understand the basic concepts but need more practice with complex problems.",
      strengths: [
        "Understands basic derivative rules",
        "Shows work clearly",
        "Attempts all problems",
      ],
      weakAreas: [
        "Struggles with chain rule applications",
        "Inconsistent with algebraic manipulation",
        "Needs more practice with word problems",
      ],
      recommendations: [
        "Focus on chain rule practice",
        "Review algebra fundamentals",
        "Work through more example problems",
      ],
    },
    detailedAnalysis: {
      conceptUnderstanding: 70,
      problemSolving: 75,
      completeness: 80,
      accuracy: 75,
    },
    studyPlan: {
      priorityTopics: [
        "Chain rule applications",
        "Algebraic manipulation",
        "Word problem setup",
      ],
      suggestedResources: [
        "Chain rule practice worksheets",
        "Algebra review materials",
        "Step-by-step problem solving guides",
      ],
      estimatedTime: "5-6 hours",
      nextSteps: [
        "Complete chain rule practice set",
        "Review algebra basics",
        "Schedule study group session",
      ],
    },
    confidence: 0.78,
  },

  poorAnalysis: {
    analysisId: "mock-poor-001",
    timestamp: "2024-01-01T00:00:00Z",
    overallScore: 45,
    feedback: {
      summary:
        "Significant improvement needed. Focus on understanding fundamental concepts before attempting complex problems.",
      strengths: [
        "Shows effort in attempting problems",
        "Basic arithmetic is correct",
      ],
      weakAreas: [
        "Fundamental understanding of derivatives",
        "Algebraic manipulation skills",
        "Problem-solving approach",
        "Mathematical notation",
      ],
      recommendations: [
        "Review basic derivative rules",
        "Practice algebra fundamentals",
        "Work with a tutor or study group",
        "Start with simpler problems",
      ],
    },
    detailedAnalysis: {
      conceptUnderstanding: 40,
      problemSolving: 35,
      completeness: 50,
      accuracy: 45,
    },
    studyPlan: {
      priorityTopics: [
        "Basic derivative rules",
        "Algebra fundamentals",
        "Function notation",
        "Graph interpretation",
      ],
      suggestedResources: [
        "Khan Academy Pre-calculus review",
        "Basic algebra practice",
        "One-on-one tutoring",
        "Study group sessions",
      ],
      estimatedTime: "10-12 hours",
      nextSteps: [
        "Schedule tutoring appointment",
        "Complete pre-calculus review",
        "Practice basic derivative rules daily",
        "Attend all office hours",
      ],
    },
    confidence: 0.65,
  },

  // Helper function to get random analysis
  getRandomAnalysis: function () {
    const analyses = [
      this.goodAnalysis,
      this.averageAnalysis,
      this.poorAnalysis,
    ];
    return analyses[Math.floor(Math.random() * analyses.length)];
  },

  // Helper function to get analysis by score range
  getAnalysisByScore: function (score) {
    if (score >= 85) return this.goodAnalysis;
    if (score >= 60) return this.averageAnalysis;
    return this.poorAnalysis;
  },
};

// Export for use in other files
if (typeof module !== "undefined" && module.exports) {
  module.exports = MockData;
}

File: backend/Models/MockData.cs
using StudentStudyAI.Models;
using System.Text.Json;

namespace StudentStudyAI.Models
{
    public static class MockData
    {
        public static AnalysisResult GetGoodAnalysis()
        {
            var feedback = new Feedback
            {
                Summary = "Excellent work! You demonstrate strong understanding of calculus concepts and show clear problem-solving skills.",
                Strengths = new List<string>
                {
                    "Clear step-by-step solutions",
                    "Correct application of derivative rules",
                    "Good use of mathematical notation",
                    "Shows understanding of chain rule"
                },
                WeakAreas = new List<string>
                {
                    "Minor calculation errors in complex problems",
                    "Could show more work for partial credit"
                },
                Recommendations = new List<string>
                {
                    "Double-check arithmetic in multi-step problems",
                    "Practice more integration techniques",
                    "Review limits and continuity concepts"
                }
            };

            var studyPlan = new StudyPlan
            {
                PriorityTopics = new List<string>
                {
                    "Integration by parts",
                    "Trigonometric integrals",
                    "Improper integrals"
                },
                SuggestedResources = new List<string>
                {
                    "Khan Academy Calculus II",
                    "Paul's Online Math Notes",
                    "Practice problems from textbook Chapter 7"
                },
                EstimatedTime = "2-3 hours per week",
                NextSteps = new List<string>
                {
                    "Complete integration practice set",
                    "Review trigonometric identities",
                    "Work on time management for exams"
                }
            };

            return new AnalysisResult
            {
                OverallScore = 92,
                FeedbackData = feedback,
                StudyPlanData = studyPlan,
                Confidence = 0.92
            };
        }

        public static AnalysisResult GetAverageAnalysis()
        {
            var feedback = new Feedback
            {
                Summary = "Good effort! You show understanding of basic concepts but need more practice with complex problems.",
                Strengths = new List<string>
                {
                    "Understands basic derivative rules",
                    "Shows work clearly",
                    "Attempts all problems"
                },
                WeakAreas = new List<string>
                {
                    "Struggles with chain rule applications",
                    "Needs more practice with integration",
                    "Algebraic manipulation errors"
                },
                Recommendations = new List<string>
                {
                    "Review chain rule thoroughly",
                    "Practice more integration problems",
                    "Work on algebraic simplification"
                }
            };

            var studyPlan = new StudyPlan
            {
                PriorityTopics = new List<string>
                {
                    "Chain rule applications",
                    "Basic integration techniques",
                    "Algebraic simplification"
                },
                SuggestedResources = new List<string>
                {
                    "Textbook Chapter 3 review",
                    "Online practice problems",
                    "Study group sessions"
                },
                EstimatedTime = "4-5 hours per week",
                NextSteps = new List<string>
                {
                    "Complete chain rule worksheet",
                    "Practice integration daily",
                    "Join study group"
                }
            };

            return new AnalysisResult
            {
                OverallScore = 75,
                FeedbackData = feedback,
                StudyPlanData = studyPlan,
                Confidence = 0.75
            };
        }

        public static AnalysisResult GetPoorAnalysis()
        {
            var feedback = new Feedback
            {
                Summary = "Needs significant improvement. Focus on understanding fundamental concepts before attempting complex problems.",
                Strengths = new List<string>
                {
                    "Shows effort in attempting problems",
                    "Basic arithmetic is correct"
                },
                WeakAreas = new List<string>
                {
                    "Fundamental concept gaps",
                    "Incorrect application of rules",
                    "Missing key steps in solutions"
                },
                Recommendations = new List<string>
                {
                    "Review basic derivative rules",
                    "Practice fundamental concepts",
                    "Seek additional help from instructor"
                }
            };

            var studyPlan = new StudyPlan
            {
                PriorityTopics = new List<string>
                {
                    "Basic derivative rules",
                    "Function composition",
                    "Limit concepts"
                },
                SuggestedResources = new List<string>
                {
                    "Khan Academy Pre-calculus review",
                    "Textbook Chapter 1-2",
                    "Office hours with instructor"
                },
                EstimatedTime = "6-8 hours per week",
                NextSteps = new List<string>
                {
                    "Schedule meeting with instructor",
                    "Complete pre-calculus review",
                    "Practice basic derivatives daily"
                }
            };

            return new AnalysisResult
            {
                OverallScore = 45,
                FeedbackData = feedback,
                StudyPlanData = studyPlan,
                Confidence = 0.45
            };
        }

        // Static properties for easy access
        public static AnalysisResult goodAnalysis => GetGoodAnalysis();
        public static AnalysisResult averageAnalysis => GetAverageAnalysis();
        public static AnalysisResult poorAnalysis => GetPoorAnalysis();
    }
}